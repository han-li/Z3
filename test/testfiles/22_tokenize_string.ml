let rec _print_string_list = function
    | [] -> ()
    | fst :: [] ->
        print_string fst; ();
    | fst :: rest ->
        print_string fst;
        print_string ",";
        _print_string_list rest;
        ()
;;

let print_string_list l = 
    print_string "[";
    _print_string_list l;
    print_string "]";
    print_newline ();
    ()
;;
let rec split_char sep str =
  try
    let i = String.index str sep in
    String.sub str 0 i ::
      split_char sep (String.sub str (i+1) (String.length str - i - 1))
  with Not_found ->
      [str];;

let l = split_char ' ' "Hello dude how are you today";;
print_string_list l;;

