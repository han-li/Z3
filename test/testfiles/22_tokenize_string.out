[Hello,dude,how,are,you,today]
